### PR TITLE
Shitmed Update 1

### PR DESCRIPTION
**Pull request by gluesniffler**
# Description

First in a series of PRs to introduce bugfixes and updates to Shitmed, this will generally feature PRs from Goobstation or Backmen as well since they are actively helping me maintain the code.

Usual Shoutouts:
Deltanedas: https://github.com/Goob-Station/Goob-Station/pull/882

---

# Changelog

:cl: Mocho, Deltanedas
- add: You can now perform surgery as a monke. Rejoice.
- add: You can perform surgery on a lot of animals now, I missed a lot of them so just ask if you want any particular critter to get it.
- tweak: Entities now perish after 60 seconds of losing their heart and/or brain.
- fix: Entities properly take asphyxiation damage after losing their brain.
- fix: Torsos being gibbable, which would break surgery or just about anything.
- fix: Items not being removed from their respective slots if the parts were gibbed rather than dropped.
- fix: Animal organs not being usable properly in surgeries
- fix: Cyborg limbs are now usable as pseudo-peg arm/legs.

### CODE REVIEW

**Review by Aidenkrz — APPROVED** [review]
I don't really like the idea of the delayed death system, I feel it should just cause damage based on missing body parts/organs/whatever. This works though.

(pretty please PR relevant changes to goob :-) )

**Comment by VMSolidus** [comment]
image

Fuck this useless goddamn test, I hate it so fucking much. Silence this hardcoded abomination. Somehow it was un-disabled during the engine update. All it does is throw a fit if a player character species has different body alerts.

**Comment by sleepyyapril** [comment]
> image
> 
> Fuck this useless goddamn test, I hate it so fucking much. Silence this hardcoded abomination. Somehow it was un-disabled during the engine update. All it does is throw a fit if a player character species has different body alerts.

Making a PR :)

**Comment by gluesniffler** [comment]
> I don't really like the idea of the delayed death system, I feel it should just cause damage based on missing body parts/organs/whatever. This works though.
> 
> (pretty please PR relevant changes to goob :-) )

Not much point in overthinking a stopgap system when woundmed with all its bells and whistles is just a week away :tm:

Will PR once I figure out how to get the OriginalBody parameter into Organs without killing TestGib.

**Comment by bruhmogus** [comment]
when this is merged will it be called badmed

**Comment by gluesniffler** [comment]
I fucking hate these bullshit tests. Off to start the goob port.

**Comment by sleepyyapril** [comment]
holy shit you fixed it

**Comment by gluesniffler** [comment]
@VMSolidus Waiter!!! Waiter!!! More LP Ops heisentests please!!!
image

File: Content.Shared/Body/Systems/SharedBodySystem.Organs.cs [hunk 1]
```diff
@@ -38,6 +40,10 @@ private void RemoveOrgan(Entity<OrganComponent> organEnt, EntityUid parentPartUi
             RaiseLocalEvent(organEnt, ref removedInBodyEv);
         }
 
+        if (TryComp(parentPartUid, out DamageableComponent? damageable)
+            && damageable.TotalDamage > 200)
```
Comment (VMSolidus) [inline Content.Shared/Body/Systems/SharedBodySystem.Organs.cs:44]:
```suggestion
            && damageable.TotalDamage > 200)
```
Need to replace the 200 here with MobThresholdSystem.TryGetThresholdForState(uid, MobState.Dead, out var deadThreshold). Don't hardcode the assumption that all player characters die when they take 200 damage, we have a codebase where that is very likely not the case.

**Review by VMSolidus — CHANGES_REQUESTED** [review]

File: Content.Shared/Body/Systems/SharedBodySystem.Organs.cs (hunk 1 above)
Comment (VMSolidus) [reply Content.Shared/Body/Systems/SharedBodySystem.Organs.cs:44]: image

Comment (gluesniffler) [reply Content.Shared/Body/Systems/SharedBodySystem.Organs.cs:44]: Oh thats handy. Though I was mostly intending for organs from badly damaged people to be useless until I get health/damageablecomponent simulations, not so much dead people.

Comment (VMSolidus) [reply Content.Shared/Body/Systems/SharedBodySystem.Organs.cs:44]: Oh wait this is not on a player character lol? Guh.

**Review by VMSolidus — APPROVED** [review]

**Pull request merged**